core/src/main/scala/Gitlab.scala

```diff
@@ -0,0 +1,318 @@
+package nelson
+
+import argonaut.Argonaut.{casecodec2, casecodec4, jencode1L}
+import argonaut.{CodecJson, DecodeJson, DecodeResult, EncodeJson}
+import nelson.Github._
+import org.http4s, http4s._
+import org.http4s.Status.Successful
+import org.http4s.argonaut.jsonOf
+import org.http4s.client.{Client, UnexpectedStatus}
+import org.http4s.headers.{Accept, Authorization, MediaRangeAndQValue}
+import org.http4s.util.CaseInsensitiveString
+import scalaz._, Scalaz._
+import scala.collection.mutable.ListBuffer
+import scalaz.concurrent.Task
+import java.net.{MalformedURLException, URI}
+
+object Gitlab {
+
+  final class GitlabHttp(cfg: ScmConfig, http4sClient: Client) extends (GithubOp ~> Task) {
+    def apply[A](in: GithubOp[A]): Task[A] = in match {
+      case GetAccessToken(fromCode: String) =>
+        fetch[AccessToken](cfg.tokenEndpoint) { uri =>
+          http4s.Request(
+            Method.POST,
+            uri
+              .withQueryParam("client_id", cfg.clientId)
+              .withQueryParam("client_secret", cfg.clientSecret)
+              .withQueryParam("code", fromCode)
+              .withQueryParam("grant_type", "authorization_code")
+              .withQueryParam("redirect_uri", cfg.redirectUri)
+          )
+        }
+
+      case GetUser(token: AccessToken) =>
+        authFetch[Github.User](cfg.userEndpoint, token)
+
+      case GetUserOrgKeys(token: AccessToken) =>
+        authFetch[List[Github.OrgKey]](cfg.userOrgsEndpoint, token)
+
+      case GetOrganizations(keys: List[Github.OrgKey], t: AccessToken) =>
+        Nondeterminism[Task].gatherUnordered(
+          keys.map(key => authFetch[Organization](cfg.orgEndpoint(key.slug), t))
+        )
+
+      case GetUserRepositories(t: AccessToken) =>
+        def go(uri: String)(accum: List[Repo]): List[Repo] = {
+          val EntityWithMeta(repos, pagination) = authFetchWithMeta[List[Repo]](uri, t).run
```

Comment:
We should never be calling `.run` in code like this.

Reply:
yeah, it's basically the same [thing](https://github.com/getnelson/nelson/blob/df7603c466aadfb76b6e75943f4e9839eafe0b54/core/src/main/scala/Github.scala#L290) as in the Github interpreter. I'll have it a look anyway.